solutions/kubernetes/ceph-deploy-functions.sh

```diff
@@ -335,17 +338,73 @@ EOF
     ceph_status
 }
 
+function install_prereq_image() {
+    add_secondary_separator "Verify/Install Docker"
+    if ! which docker; then
+        if [[ "$ID" == "rocky"  ]]; then
+            dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
+            dnf install -y docker-ce docker-ce-cli containerd.io
+            check_status "$HOSTNAME: Docker installation failed"
+            systemctl enable docker && systemctl start docker
+        fi
+        
+        if [[ "$ID" == "ubuntu" || "$ID" == "centos" ]]; then
+            pushd $SCRIPTS_LOCATION
+                curl -fsSL https://get.docker.com -o get-docker.sh
+                chmod +x get-docker.sh
+                ./get-docker.sh
+                check_status "$HOSTNAME: Docker installation failed"
+            popd
+        fi
+    fi
+
+    add_secondary_separator "Install Cephadm"
+    pushd $SCRIPTS_LOCATION
+        curl --silent --remote-name --location https://github.com/ceph/ceph/raw/quincy/src/cephadm/cephadm && chmod +x cephadm
+        unlink /usr/local/bin/cephadm 
+        ln -s $SCRIPTS_LOCATION/cephadm /usr/local/bin/cephadm
+    popd
+
+    add_secondary_separator "Pull Ceph Docker image"
+    docker pull $CEPH_IMAGE
+}
+
+function deploy_ceph_image() {
+    add_secondary_separator "Deploy Ceph"
+    cephadm --image $CEPH_IMAGE --verbose bootstrap --mon-ip $(hostname -i) --initial-dashboard-user admin --initial-dashboard-password cephadmin --dashboard-password-noupdate --single-host-defaults --skip-pull --skip-monitoring-stack --allow-fqdn-hostname --allow-overwrite
+}
+
 function io_operation() {
-    add_secondary_separator "Add RADOS-GW User"
-    radosgw-admin user create --uid=io-test --display-name="io-ops"
+    if [[ $CEPH_DOCKER_DEPLOYMENT = "true" ]]; then
+        add_secondary_separator "Add RADOS-GW User"
+        cephadm shell -- radosgw-admin user create --uid=io-test --display-name="io-ops"
 
-    add_secondary_separator "Setup Dashboard RADOS User"
-    ceph dashboard set-rgw-credentials
+    elif [[ $CEPH_DOCKER_DEPLOYMENT = "false" ]]; then
+        add_secondary_separator "Add RADOS-GW User"
+        radosgw-admin user create --uid=io-test --display-name="io-ops"
+
+        add_secondary_separator "Setup Dashboard RADOS User"
+        ceph dashboard set-rgw-credentials
+
+    else
+        echo "Not ceph deployment."
+    fi
 
     pushd /var/tmp/
         ./io-sanity.sh
```

Comment:
should we check status of `io-sanity.sh` script by calling check_status function? or it will gracefully exit on any error?

Reply:
In io-sanity each command has check_status after execution and prints errors and exits out on any problem. I think pos script execution status check is not required here maybe.
Please share your thoughts @shailesh-vaidya.

Reply:
Added status check on io script execution.